python/generateTemplate.py

```diff
@@ -0,0 +1,1155 @@
+"""Attempts to create airmass-matched template images for existing images in an LSST repository."""
+
+# LSST Data Management System
+# Copyright 2016 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import print_function, division, absolute_import
+from collections import namedtuple
+
+import numpy as np
+from scipy import constants
+from scipy.ndimage.interpolation import shift as scipy_shift
+from scipy.ndimage.morphology import binary_dilation
+
+from lsst.afw.coord import Coord, IcrsCoord
+import lsst.afw.geom as afwGeom
+from lsst.afw.geom import Angle
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+from lsst.daf.base import DateTime
+import lsst.daf.persistence as daf_persistence
+import lsst.meas.algorithms as measAlg
+# import lsst.pex.exceptions
+import lsst.pex.policy as pexPolicy
+from lsst.sims.photUtils import Bandpass
+from lsst.utils import getPackageDir
+
+from .lsst_defaults import lsst_observatory, lsst_weather
+from .dcr_utils import calculate_rotation_angle
+from .dcr_utils import diff_refraction
+from .dcr_utils import solve_model
+from .dcr_utils import wrap_warpExposure
+
+__all__ = ['GenerateTemplate']
+
+nanFloat = float("nan")
+nanAngle = Angle(nanFloat)
+
+# Temporary debugging parameters, used if debug_mode=True or self.debug=True is set.
+# In debug mode, the DCR model is only calculated for pixels within [y0: y0 + dy, x0: x0 + dx]
+x0 = 300
+dx = 200
+y0 = 500
+dy = 200
+
+
+class GenerateTemplate:
+    """Lightweight object with only the minimum needed to generate DCR-matched template exposures.
+
+    A model must first be generated with DcrCorrection (below). That model can then be used directly, or
+    persisted and later read back in. GenerateTemplate requires much less memory than DcrCorrection,
```

Comment:
Looks like I missed a few places when I was changing the names. I've fixed that and changed the wording a bit to hopefully be more explicit.